cli/commands.go

```diff
@@ -24,6 +24,37 @@ import (
 )
 
 const defaultRegistry = "docker.io"
+const clusteNameMaxSize int = 35
+
+// Make sure a cluster name is also a valid host name according to RFC 1123.
+// We further restrict the length of the cluster name to shorter than 'clusterNameMaxSize'
+// so that we can construct the host names based on the cluster name, and still stay
+// within the 64 characters limit.
+func checkClusterName(name string) error {
+       if len(name) > clusteNameMaxSize {
+	       return fmt.Errorf("cluster name is too long")
+       }
+
+       if name[0] == '-' || name[len(name) - 1] == '-' {
+	       return fmt.Errorf("cluster name can not start or end with - (dash)")
+       }
+
+
+       for _ , c := range name {
```

Comment:
How does this compare to a regexp matching like
```Go
regex := regexp.MustCompile(`"^(([a-zA-Z]|[a-zA-Z][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z]|[A-Za-z][A-Za-z0-9\-]*[A-Za-z0-9])$"`)
ok := regex.MatchString(name)
```
?
Probably your solution is easier to read :thinking: 
(Regex is from https://stackoverflow.com/a/106223/6450189 )

Reply:
Yes, I think so.